mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```
@@ -2895,12 +2905,12 @@ struct GridwiseGemmAccelRewritePattern
     // Alocate LDS and create subviews.
 
     // Compute required LDS sizes.
-    int64_t ldsBlockASize = kpacksPerBlock * mPerBlock * kpack;
-    int64_t ldsBlockBSize = kpacksPerBlock * nPerBlock * kpack;
-    LLVM_DEBUG(llvm::dbgs()
-               << "LDS block sizes (bytes): "
-               << ldsBlockASize * getByteWidth(elementTypeA) << " "
-               << ldsBlockBSize * getByteWidth(elementTypeB) << "\n");
+    int64_t ldsBlockASize =
+        getPackedByteSize(kpacksPerBlock * mPerBlock * kpack, elementTypeA);
+    int64_t ldsBlockBSize =
+        getPackedByteSize(kpacksPerBlock * nPerBlock * kpack, elementTypeB);
+    LLVM_DEBUG(llvm::dbgs() << "LDS block sizes (bytes): " << ldsBlockASize
+                            << " " << ldsBlockBSize << "\n");
     if (failed(checkLDSSize(op, ldsBlockASize, ldsBlockBSize)))
```

This is where it checks for whether it fits into LDS or not. 

notice earlier, it was passing `ldsBlockASize` which is just number of elements

But 

https://github.com/ROCm/rocMLIR/blob/bb1be98a7a8393cb49b9583c2003bfeb1f0d2933/mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp#L253

Expects "bytes"

---

yes, I introduced this bug in the refactor I think it was ok previously.

---

this check is technically not needed, because we'll do it properly in reuseLDS after figuring out what we can reuse. However, we can make it fail earlier here and save some time

---

Better to keep to fail early IMO. 
Or else we will have to remove it from all the AttentionAccel, GemmAccel lowering too for consistency.